app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -465,26 +467,39 @@ private static void recordSuccessfulSyncTime(String username) {
 
     //TODO: This and the normal sync share a ton of code. It's hard to really... figure out the right way to 
     private Pair<Integer, String> recover(HttpRequestEndpoints requestor, AndroidTransactionParserFactory factory) {
-        this.publishProgress(PROGRESS_RECOVERY_NEEDED);
-
-        Logger.log(AndroidLogger.TYPE_USER, "Sync Recovery Triggered");
+        while (asyncRestoreHelper.retryWaitPeriodInProgress()) {

```

Comment:
It will be full seconds (many of them) for the retry period, so probably 500ms or so?

-Clayton